sdk/go/pulumi/types.go

```diff
@@ -717,7 +735,7 @@ func awaitInputs(ctx context.Context, v, resolved reflect.Value) (bool, bool, []
 			valueType = v.Type()
 		} else {
 			// Handle pointer inputs.
-			if v.Kind() == reflect.Ptr {
+			if v.Kind() == reflect.Ptr && !(v.CanInterface() && v.Type().Implements(resourceType)) {
```

Comment:
Merge master, I've just changed this same bit of code to handle the case of v being a pointer but resolved being interface{}